cluster-logging.md

```diff
@@ -0,0 +1,110 @@
+# Cluster Logging and OpenTelemetry
+
+This is the protocol and semantic conventions documentation for Red Hat OpenShift Logging.
+
+## Forwarding and Ingestion Protocol
+
+Red Hat OpenShift Logging provides a log collection and forwarding solution that is capable of writing logs to OpenTelemetry endpoints using OTLP. [OTLP](https://opentelemetry.io/docs/specs/otlp/) is the *protocol* for encoding, transporting, and delivering telemetry data.  This product additionally is capable of providing a Loki storage deployment that provides an OTLP endpont to ingest log streams.  This document defines the semantic conventions associated with the logs collected from the various sources of an OpenShift cluster.
+
+**Note:** Logs are forwarded using OTLP/HTTP as defined by the OpenTelemetry Observability Framework.  It uses Protobuf payloads encoded in JSON format.
+
+
+## Semantic Conventions
+
+The log collector provided by this solution collects the following log streams:
+
+* Container logs
+* Cluster node journal logs
+* Cluster node auditd logs 
+* Kubernetes and OpenShift API server logs
+* OpenShift Virtual Network (OVN) logs
+
+These streams are forwarded using the sementatic conventions defined by [OpenTelemetry semantic attributes](https://github.com/open-telemetry/semantic-conventions/tree/main/docs). The semantic conventions in OpenTelemetry define a *Resource* as an immutable representation of the entity producing telemetry as *Attributes*. For example, a process producing telemetry that is running in a container has a container_name, a cluster_id, a pod_name, a namespace, and possibly a deployment or app_name. All of these *Attributes* are included in the *Resource* object.  This grouping and reducing of common attributes is a powerful tool when sending logs as telemetry data.
+
+
+The following sections define the attributes that are forwarded.
+
+
+### Common Attributes
+
+All log streams include a minimal set of attributes.
+
+Resource Attributes:
+```
+cluster.id
+node.name
+openshift.log.source
+openshift.log.type
+```
+
+Log Attributes:
```

Comment:
By my reading of https://opentelemetry.io/docs/specs/otel/logs/data-model/#log-and-event-record-definition
these are the top-level log "fields", which are unique to log signals. All the other attributes go under the "attributes" field, and are common across signals (all signals have an "attributes" field). Doesn't change this proposal but worth noting.

Reply:
> these are the top-level log "fields", which are unique to log signals. All the other attributes go under the "attributes" field

@alanconway  What do you mean by "top level fields"?  How is it should we distinguish them in the document from "resource" or "log level" fields?  I'm actually still struggling to justify the need to distinguish the level at which these attributes are present.  From my perspective of a consumer, the "level" seems irrelevent; they are simply "attributes" of a signal.

Reply:
Updating text to identify the minimal fields separate from the set of attributes